openfl/component/director/director.py

```diff
@@ -172,38 +171,37 @@ def remove_experiment_data(self, experiment_name: str, caller: str):
         if experiment_name in self.experiment_stash.get(caller, {}):
             del self.experiment_stash[caller][experiment_name]
 
-    def collaborator_health_check(
-            self, *, collaborator_name: str, is_experiment_running: bool
+    def envoy_health_check(
+            self, *, envoy_name: str, is_experiment_running: bool,
+            cuda_devices_status: list = None,
     ) -> int:
         """Accept health check from envoy."""
-        shard_info = self._shard_registry.get(collaborator_name)
+        shard_info = self._shard_registry.get(envoy_name)
         if not shard_info:
-            raise Exception(f'Unknown shard {collaborator_name}')
+            raise Exception(f'Unknown shard {envoy_name}')
 
         hc_period = self.settings.get('envoy_health_check_period', ENVOY_HEALTH_CHECK_PERIOD)
         shard_info['is_online']: True
         shard_info['is_experiment_running'] = is_experiment_running
         shard_info['valid_duration'] = 2 * hc_period
         shard_info['last_updated'] = time.time()
 
+        if cuda_devices_status is not None:
+            for i in range(len(cuda_devices_status)):
+                shard_info['shard_info']['node_info']['cuda_devices'][i] = cuda_devices_status[i]
```

Comment:
`cuda_devices` is `repeated` field, so it should be used as a `list` not a `dict`.
```python
 shard_info['shard_info']['node_info']['cuda_devices'].extend(cuda_devices_status)
```